EpsilonLights/Src/Lights.c

```diff
@@ -7,90 +7,89 @@ void updateLightsTask(void const* arg)
     // One time osDelayUntil intialization
     uint32_t prevWakeTime = osKernelSysTick();
     // Store inputs values
-    char headlightsOff;
-    char headlightsLow;
-    char headlightsHigh;
-    char rightSignal;
-    char leftSignal;
-    char hazards;
-    char brakes;
-    char bmsStrobe;
-    uint32_t regenBrakeInt;
-    float regenBrakeFloat;
+
+
+    lightsInfo lightsCharacteristics;
+
 
     // NOTE: All Lights Out pins are active low
     for (;;)
     {
-        regenBrakeInt = 0;
-        regenBrakeFloat = 0;
-
-        osDelayUntil(&prevWakeTime, LIGHTS_UPDATE_FREQ);
-        headlightsOff = (lightsInputs >> HOFF_INPUT_INDEX) & 1;
-        headlightsLow = (lightsInputs >> HLOW_INPUT_INDEX) & 1;
-        headlightsHigh = (lightsInputs >> HHIGH_INPUT_INDEX) & 1;
-        rightSignal = (lightsInputs >> RSIGNAL_INPUT_INDEX) & 1;
-        leftSignal = (lightsInputs >> LSIGNAL_INPUT_INDEX) & 1;
-        hazards = (lightsInputs >> HAZARDS_INPUT_INDEX) & 1;
-        bmsStrobe = ((auxBmsInputs[1] >> 0) & STROBE_FAULT_MASK) & 1;
-        regenBrakeInt |= (driversInputs[1] & REGEN_BRAKE_MASK_1) >> 4;
-        regenBrakeInt |= (driversInputs[2] & REGEN_BRAKE_MASK_2) << 4;
-
-        /* UPDATE HEADLIGHTS */
-        if ((headlightsOff))
-        {
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_OFF);
-        }
-        else if ((headlightsLow && headlightsHigh))
-        {
-            // Error state, turn only the low headlights on.
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_ON);
-        }
-        else
-        {
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, headlightsHigh);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, headlightsLow);
-        }
+        updateLights1 (&lightsCharacteristics, &prevWakeTime);
+    }
+}
 
-        /* UPDATE BRAKE LIGHTS */
-        brakes = (driversInputs[BRAKES_INPUT_INDEX_P1] >> BRAKES_INPUT_INDEX_P2) & 1;
-        regenBrakeFloat = (float)regenBrakeInt;
+void updateLights1 (lightsInfo* lightsCharacteristics, uint32_t* prevWakeTime)
+{
+    lightsCharacteristics -> regenBrakeInt = 0;
```

Comment:
Just a stylistic thing but I would change `lightsCharacteristics -> regenBrakeInt` into `lightsCharacteristics->regenBrakeInt`. That goes for every struct pointer dereferencing.